src/main/java/me/magnet/consultant/ServiceInstanceBackend.java

```diff
@@ -31,14 +85,67 @@
 	 * @param consulUri    The URI where Consul's API can be found.
 	 * @param objectMapper The ObjectMapper which can be used to deserialize JSON.
 	 * @param http         The HTTP client to use.
+	 * @param cacheLocateCallsForMillis How long the results of locate calls should be cached for.
 	 */
 	ServiceInstanceBackend(Optional<String> datacenter, URI consulUri, ObjectMapper objectMapper,
-			CloseableHttpClient http) {
+			CloseableHttpClient http, long cacheLocateCallsForMillis) {
 
 		this.datacenter = datacenter;
-		this.consulUri = consulUri;
-		this.objectMapper = objectMapper;
-		this.http = http;
+
+		this.serviceInstances = CacheBuilder.newBuilder()
+				.expireAfterWrite(cacheLocateCallsForMillis, TimeUnit.MILLISECONDS)
+				.build(new CacheLoader<ServiceIdentifierCacheKey, List<ServiceInstance>>() {
```

Comment:
You could do `CacheLoader.from(Function)` as well here

Reply:
Fixed, thanks!